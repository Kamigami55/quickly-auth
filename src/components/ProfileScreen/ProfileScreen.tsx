import { CircleUser } from 'lucide-react';

import { Avatar, AvatarFallback } from '@/components/ui/avatar';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Profile } from '@/types/profile';

export function ProfileScreen({ profile }: { profile: Profile }) {
  return (
    <div className="flex flex-col gap-4">
      <Card>
        <CardHeader>
          <CardTitle>User</CardTitle>
        </CardHeader>
        <CardContent>
          <div className="flex items-center gap-4">
            <Avatar className="hidden h-9 w-9 sm:flex">
              <AvatarFallback>
                <CircleUser className="h-5 w-5" />
              </AvatarFallback>
            </Avatar>
            <div className="grid gap-1">
              <p className="text-sm font-medium leading-none">
                {profile?.user?.first_name} {profile?.user?.last_name}
              </p>
              <p className="text-sm text-muted-foreground">
                {profile?.user?.email}
              </p>
            </div>
          </div>
        </CardContent>
      </Card>

      <Card>
        <CardHeader>
          <CardTitle>Company</CardTitle>
        </CardHeader>
        <CardContent className="grid gap-4">
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500">Legal Name:</div>
            <div className="text-sm">{profile?.company?.legal_name || '-'}</div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500">Business Type:</div>
            <div className="text-sm">
              {profile?.company?.business_type || '-'}
            </div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500">Industry:</div>
            <div className="text-sm">{profile?.company?.industry || '-'}</div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500">Website:</div>
            <div className="text-sm">{profile?.company?.website || '-'}</div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500">Business Registration:</div>
            <div className="text-sm">
              {profile?.company?.business_registration || '-'}
            </div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500">Phone:</div>
            <div className="text-sm">{profile?.company?.phone || '-'}</div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500">Business Number:</div>
            <div className="text-sm">
              {profile?.company?.business_number || '-'}
            </div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500 ">Expected Activity:</div>
            <div className="text-sm">
              {profile?.company?.expected_activity || '-'}
            </div>
          </div>
          <div className="grid grid-cols-2 gap-2">
            <div className="text-sm text-gray-500 ">Early Pay Intent:</div>
            <div className="text-sm">
              {profile?.company?.early_pay_intent ? 'Yes' : 'No'}
            </div>
          </div>
        </CardContent>
      </Card>
    </div>
  );
}
