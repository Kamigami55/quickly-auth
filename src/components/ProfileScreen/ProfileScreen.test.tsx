import '@testing-library/jest-dom';

import { render } from '@testing-library/react';

import { ProfileScreen } from './ProfileScreen';

const profile = {
  user: {
    first_name: 'Eason',
    last_name: 'Chang',
    email: 'eason@example.com',
    Company: {
      legal_name: 'ACME Inc.',
      business_registration: 'corporate',
      business_type: 'Digital products',
      industry: 'Apps',
      expected_activity: 'Get my invoices paid early',
      early_pay_intent: 1,
      website: 'https://acme.com',
      business_number: '123456789',
      phone: '1234567890',
      has_trade_name: false,
    },
  },
  company: {
    legal_name: 'ACME Inc.',
    business_registration: 'corporate',
    business_type: 'Digital products',
    industry: 'Apps',
    expected_activity: 'Get my invoices paid early',
    early_pay_intent: 1,
    website: 'https://acme.com',
    business_number: '123456789',
    phone: '1234567890',
    has_trade_name: false,
  },
};

describe('ProfileScreen', () => {
  it('renders profile screen unchanged', () => {
    const { container } = render(<ProfileScreen profile={profile} />);
    expect(container).toMatchSnapshot();
  });
});
