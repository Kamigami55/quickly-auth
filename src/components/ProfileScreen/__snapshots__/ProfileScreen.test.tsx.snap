// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen renders profile screen unchanged 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="text-2xl font-semibold leading-none tracking-tight"
        >
          User
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <div
          class="flex items-center gap-4"
        >
          <span
            class="relative shrink-0 overflow-hidden rounded-full hidden h-9 w-9 sm:flex"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-muted"
            >
              <svg
                class="lucide lucide-circle-user h-5 w-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
                <path
                  d="M7 20.662V19a2 2 0 0 1 2-2h6a2 2 0 0 1 2 2v1.662"
                />
              </svg>
            </span>
          </span>
          <div
            class="grid gap-1"
          >
            <p
              class="text-sm font-medium leading-none"
            >
              Eason
               
              Chang
            </p>
            <p
              class="text-sm text-muted-foreground"
            >
              eason@example.com
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="text-2xl font-semibold leading-none tracking-tight"
        >
          Company
        </h3>
      </div>
      <div
        class="p-6 pt-0 grid gap-4"
      >
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500"
          >
            Legal Name:
          </div>
          <div
            class="text-sm"
          >
            ACME Inc.
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500"
          >
            Business Type:
          </div>
          <div
            class="text-sm"
          >
            Digital products
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500"
          >
            Industry:
          </div>
          <div
            class="text-sm"
          >
            Apps
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500"
          >
            Website:
          </div>
          <div
            class="text-sm"
          >
            https://acme.com
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500"
          >
            Business Registration:
          </div>
          <div
            class="text-sm"
          >
            corporate
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500"
          >
            Phone:
          </div>
          <div
            class="text-sm"
          >
            1234567890
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500"
          >
            Business Number:
          </div>
          <div
            class="text-sm"
          >
            123456789
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500 "
          >
            Expected Activity:
          </div>
          <div
            class="text-sm"
          >
            Get my invoices paid early
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="text-sm text-gray-500 "
          >
            Early Pay Intent:
          </div>
          <div
            class="text-sm"
          >
            Yes
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
